### DISCUSSION TITLE
SpinalTag Question

### DISCUSSION THREAD

**Discussion by openpowerwtf** [SpinalHDL/SpinalHDL#1255] [Q&A] [answered]
I am able to create a custom tag and add it to a signal/find after elaboration, but don't see a way for the tag to reference its parent; i.e. can the tag do a ```setName()```, ```addAttribute()```, etc. on the signal it's being added to?

**Comment by andreasWallner** [comment, marked as answer]
A tag has no inherent method of finding the signal/component it's on (you could add the same tag instance on multiple components if you wanted to). If you need that then you have to pass the signal/component to the tag at some point (or have the tag search itself...).

SpinalHDL uses Tags just as a marker/carrier of some additionally needed information. The logic itself is implemented by having a pass (could also be some rework step) that uses the tags as information on what to do. The tag itself is only carrier the additional information needed.

**Reply by openpowerwtf** [reply]
Thanks - I think that is what I have done.  I added a phase to search for the tag using walkDeclarations and then modify the signal.  Seems to work except when in ClockingArea, so I may have to modify my search.
